### PR TITLE
EVA-2364 download single sequence

### PR DESCRIPTION
**Pull request by tcezard**
New class for downloading single sequences from NCBI

### CODE REVIEW

File: ebi_eva_common_pyutils/assembly/assembly.py [hunk 1]
```diff
@@ -11,231 +11,5 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import os
```
Comment (tcezard) [inline ebi_eva_common_pyutils/assembly/assembly.py:14]: This content was moved wholesale to `ebi_eva_common_pyutils/reference/assembly.py`

**Review by sundarvenkata-EBI — APPROVED** [review]

File: ebi_eva_common_pyutils/reference/sequence.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        self.species_scientific_name = species_scientific_name
+        self.reference_directory = reference_directory
+        self.eutils_api_key = eutils_api_key
+
+    @classmethod
+    def is_genbank_accession_format(cls, accession):
+        if any(
+                re.match(insdc_accession_format, accession)
+                for insdc_accession_format in cls.insdc_accession_formats
+        ):
```
Comment (apriltuesday) [inline ebi_eva_common_pyutils/reference/sequence.py:59]: This isn't very important but you could do `'|'.join(insdc_accession_format)` and then just do a single regex match.

Comment (tcezard) [reply ebi_eva_common_pyutils/reference/sequence.py:59]: I know but I find the `any` syntax more readable than the or and the cost of creating the different match object seem negligible.

File: ebi_eva_common_pyutils/reference/assembly.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        self.check_assembly_accession_format(assembly_accession)
+        self.assembly_accession = assembly_accession
+        self.species_scientific_name = species_scientific_name
+        self.reference_directory = reference_directory
+        self.eutils_api_key = eutils_api_key
+
+    @staticmethod
+    def is_assembly_accession_format(assembly_accession):
+        if re.match(r"^GC[F|A]_\d+\.\d+$", assembly_accession) is None:
+            return False
```
[the remark this answers is not in the document]
Comment (tcezard) [reply ebi_eva_common_pyutils/reference/assembly.py:51]: That's a good point I actually split that function and didn't do a good job of it. I fixed that and added tests.

File: ebi_eva_common_pyutils/reference/assembly.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+    @staticmethod
+    def check_assembly_accession_format(assembly_accession):
+        if NCBIAssembly.is_assembly_accession_format(assembly_accession):
+            raise ValueError('Invalid assembly accession: it has to be in the form of '
+                            'GCF_XXXXXXXXX.X or GCA_XXXXXXXXX.X where X is a number')
```
[the remark this answers is not in the document]
Comment (tcezard) [reply ebi_eva_common_pyutils/reference/assembly.py:57]: changed it.

File: ebi_eva_common_pyutils/reference/sequence.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        r'[A-Z]{6}[0-9]{9}\.[0-9]+'
+    ]
+
+    def __init__(self, sequence_accession, species_scientific_name, reference_directory, eutils_api_key=None):
+        self.sequence_accession = sequence_accession
+        self.species_scientific_name = species_scientific_name
+        self.reference_directory = reference_directory
+        self.eutils_api_key = eutils_api_key
+
+    @classmethod
```
[the remark this answers is not in the document]
Comment (tcezard) [reply ebi_eva_common_pyutils/reference/sequence.py:54]: Done

**Review by apriltuesday — APPROVED** [review]

**Pull request merged**